lib/web_ui/test/canvaskit/common.dart

```diff
@@ -189,6 +190,23 @@ Future<void> matchPictureGolden(String goldenFile, CkPicture picture,
       region: region, maxDiffRatePercent: 0.0, write: write);
 }
 
+Future<bool> matchImage(ui.Image left, ui.Image right) async {
+  if (left.width != right.width || left.height != right.height) {
+    return false;
+  }
+  int getPixel(ByteData data, int x, int y) => data.getUint32((x + y * left.width) * 4);
+  final ByteData leftData = (await left.toByteData())!;
+  final ByteData rightData = (await right.toByteData())!;
+  for (int y = 0; y < left.height; y++) {
+    for (int x = 0; x < left.width; x++) {
+      if (getPixel(leftData, x, y) != getPixel(rightData, x, y)) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
```

Comment:
This is a copy of: https://github.com/flutter/engine/blob/3cfc77dc0c2ac5800b4dc5ba802e0e66d57c0010/testing/dart/canvas_test.dart#L116-L133

Let me know if there's an easy way to deduplicate this code!